javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,189 @@
+"use strict";
+
+//Creating global scope variables
+let userName = "";
+let todos = []; //declared through let to be able to modify the array through the filter method
+
+//Creating a function
+function getReply(command) {
+  let commandSplitted = command.split(" "); //Splits the command into separate words for easy search through them
+
+  //Name actions
+  //.toLowerCase is used to get rid of case sensitivity
+  if (
+    command.toLowerCase().includes("my name is") ||
+    command.toLowerCase().includes("i am")
+  ) {
+    let name = commandSplitted[commandSplitted.length - 1];
+    if (name == userName) {
+      return `Hi ${userName}! Glad to see you again`;
+    } else {
+      userName = name;
+      return `Nice to meet you ${name}`;
+    }
+  } else if (command.toLowerCase().includes("is my name")) {
+    if (userName == undefined || userName == "") {
+      return "I haven't met you yet. Please introduce yourself";
+    } else {
+      return userName;
+    }
```

Comment:
This works great, but you could use the "Return Early" technic. (no need for the last else statement)
If you have time, you could take a look at it: https://dev.to/jenniferlynparsons/early-returns-in-javascript-5hfb

Reply:
Thanks! I will look into it :)